### DISCUSSION TITLE
Sintering two particles

### DISCUSSION THREAD

**Discussion by Judy-22** [idaholab/moose#20805] [Q&A Modules: Phase field] [answered]
Hi @SudiptaBiswas,

I am trying to reproduce the baseline case presented in your article https://link.springer.com/article/10.1007/s10853-017-1846-3   (table 2, case I).

For time less than 6 it seems perfect.
t = 1s
![t_1](https://user-images.githubusercontent.com/78077005/163666545-559f0948-6339-405b-82a3-d12cd0fbad61.png)
t = 6
![t_6](https://user-images.githubusercontent.com/78077005/163666572-6b40de45-d4c0-4767-b3e2-5606f863f657.png)

But when the time is greater than 6, the time step reduces drastically, and tends to reduce more and more, which causes convergence problems.

See the result:

![t_6 11](https://user-images.githubusercontent.com/78077005/163666768-5881a265-eef0-4e97-af9d-1fe8924f8388.png)
![t_6 12](https://user-images.githubusercontent.com/78077005/163666777-93241d0d-6c86-45ea-8267-261952b35457.png)

_____________________________________________________________________________

The input file:
```
 ## The phase field model to simulate sintering of two particles in 2D

[GlobalParams]        # Automatically define order parameters 
  var_name_base = eta
  op_num = 2
[]

[Mesh]                # Generate mesh
  type = GeneratedMesh
  dim = 2
  nx = 80             # n°s element in x. 
  ny = 40             # n°s element in y.
  xmin = 0  
  xmax = 40      # µm
  ymin = 0
  ymax = 20      # µm
  elem_type = QUAD4
  uniform_refine = 2
[]

[Variables]
  [./c]               # particle concentration (conserved variable)
    order = FIRST
    family = LAGRANGE
  [../]
  [./w]               # functional derivative of the total free energy  
    order = FIRST
    family = LAGRANGE
  [../]
  [./PolycrystalVariables] # Automatically creates order parameter variables
  [../]
[]

[Kernels]
  [./c_res]
  # compute the residual therm of total free energy as function of c.
    type = SplitCHParsed
    variable = c
    f_name = F
    kappa_name = kappa_c
    args = 'eta0 eta1'
    w = w
  [../]
  [./w_res]
  # compute the residual therm of functional free energy as function of w and M.
    type = SplitCHWRes
    variable = w
    mob_name = M
  [../]
  [./time]
  # compute the time derivative term of w and c.
    type = CoupledTimeDerivative
    variable = w
    v = c
  [../]
  [./eta0_dot]
    type = TimeDerivative
    variable = eta0
  [../]
  [./acint_eta0]
    type = ACInterface
    variable = eta0
    mob_name = L
    args = c
    kappa_name = kappa_eta
  [../]
  [./acbulk_eta0]
    type = AllenCahn
    variable = eta0
    mob_name = L
    f_name = F
    args = 'c eta1'
  [../]
  [./eta1_dot]
    type = TimeDerivative
    variable = eta1
  [../]
  [./acint_eta1]
    type = ACInterface
    variable = eta1
    mob_name = L
    args = c
    kappa_name = kappa_eta
  [../]
  [./acbulk_eta1]
    type = AllenCahn
    variable = eta1
    mob_name = L
    f_name = F
    args = 'c eta0'
  [../]
[]

[Materials]  
  [./constants]
    type = GenericFunctionMaterial
    prop_names = 'kappa_c   kappa_eta     A      B     L
                    Dvol      Dvap       Dsuf    Dgb'
    
    prop_values = '1.0       0.5      16.0     1.0    1.0
                   0.01       0.001     4.0     0.4'
  [../]
  [./free_energy]            # Total free energy function of c and eta.
    type = DerivativeParsedMaterial
    f_name = F
    args = 'c eta0 eta1'          # Must be changed as op_num changes.
    function = 'A*c^2*(1-c)^2 + B*(c^2+6*(1-c)*(eta0^2+eta1^2)-4*(2-c)*(eta0^3+eta1^3)
                + 3*(eta0^2+eta1^2)^2)'
    material_property_names = 'A  B'
    derivative_order = 2
  [../]
  [./CHmobility]    # Cahn and Hilliard Mobility incluing all diffusin path (m^5/J/s)
    type = DerivativeParsedMaterial
    f_name = M
    args = 'c eta0 eta1'
    function = '(Dvol*c^3*(10 - 15*c + 6*c^2) + Dvap*(1 - (c^3*(10 - 15*c + 6*c^2)))
                + Dsuf*(1 - c) + Dgb*(eta0*eta1))'
    material_property_names = 'Dvol Qvol Dvap Qvap Dsuf Qs Dgb Qgb KB Vm d2F:=D[F(c),c,c] T'
    derivative_order = 1
  [../]
[]

[BCs]
  [./bcs]
    #zero flux BC
    type = NeumannBC
    value = 0
    variable = c
    boundary = '0 1 2 3'
  [../]
[]

[AuxVariables]
  [./bnds]
  # Auxivariables that define the location of grain boundaries in a polycrystalline sample
  [../]
  [./free_energy]
    order = CONSTANT
    family = MONOMIAL
  [../]
[]

[AuxKernels]
  [./bnds]
  # Calculate location of grain boundaries in a polycrystalline sample
    type = BndsCalcAux
    variable = bnds
    var_name_base = eta
    op_num = 2
    v = 'eta0 eta1'
  [../]
  [./energy_density]
    type = TotalFreeEnergy
    variable = free_energy
    f_name = F
    kappa_names = 'kappa_c kappa_eta kappa_eta'
    interfacial_vars = 'c eta0 eta1'
  [../]
[]

[ICs]
  [./ic_eta0]
  # initial condition for each order parameter 
    type = SmoothCircleIC
    variable = eta0
    x1 = 10.5
    y1 = 10.0
    radius = 7.5
    invalue = 1.0
    outvalue = 0.0
    int_width = 2.0
  [../]
  [./IC_eta1]
    type = SmoothCircleIC
    variable = eta1
    x1 = 25.5
    y1 = 10.0
    radius = 7.5
    invalue = 1.0
    outvalue = 0.0
    int_width = 2.0
  [../]
  [./ic_c]
  # initial condition for conserved parameter 
    type = SpecifiedSmoothCircleIC
    invalue = 1.0
    outvalue = 0.0
    int_width = 2.0
    x_positions = '10.5 25.5'
    y_positions = '10.0 10.0'
    z_positions = '0.0 0.0'
    radii       = '7.5 7.5'
    3D_spheres = false
    variable = c
    block = 0
  [../]
[]

[Preconditioning]
  [./coupled]
    type = SMP
    full = true
  [../]
[]

[Executioner]
  type = Transient
  scheme = bdf2
  solve_type = NEWTON
  automatic_scaling = true
  petsc_options_iname = '-pc_type -ksp_gmres_restart -sub_ksp_type
                         -sub_pc_type -pc_asm_overlap'
  petsc_options_value = 'asm      31                  preonly
                         lu          1'
  l_tol = 1e-05
  nl_max_its = 30
  l_max_its = 30
  nl_rel_tol = 1e-07
  nl_abs_tol = 1e-09
  start_time = 0.0
  end_time = 100
  #dt = 0.05
  [./TimeStepper]
    type = IterationAdaptiveDT
    cutback_factor = 0.5
    dt = 0.01
    growth_factor = 2
    optimal_iterations = 8
  [../]
  [./Adaptivity]
    initial_adaptivity = 1
    refine_fraction = 0.6
    coarsen_fraction = 0.01
    max_h_level = 2
    print_changed_info = true
  [../]
[]

[Outputs]
  exodus = true
  [./debug]
    type = VariableResidualNormsDebugOutput
  [../]
[]
```

**Comment by chongxiyuan** [comment, marked as answer]
Hi all,

I am also tyring to work sintering in MOOSE, and I am using a very similar input file with AC-CH kernels and free energy(following MOOSE examples and forum). I see the same issue when when the diffusivity(CH mobility) is dependent on the conserved(rho) and non-conserved(eta) phase field variables. The simulation diverges with huge vibrations in the solution of rho. When I simplify the diffusivity to a constant (D=1), the solution is fine and no more vibration in the end.

D dependent on etas and rho, tstep=1
![image](https://user-images.githubusercontent.com/35067395/166289609-9a3463f9-93b7-4ad8-8a15-b4abf657d288.png)
D dependent on etas and rho, tstep=63
![image](https://user-images.githubusercontent.com/35067395/166289650-f6797845-96eb-4b81-b17e-c3cb245e73f8.png)

input file used attached

[input.txt.txt](https://github.com/idaholab/moose/files/8604835/input.txt.txt)

D=1 constant, tstep=2400
![image](https://user-images.githubusercontent.com/35067395/166289810-1b949eb1-f579-42ce-8784-90c0351fe0dc.png)

I am suspecting maybe the element order is not high enough, or mesh is not refined enough for the phase field variables? Any help or discussion is welcomed. Thank you.

**Reply by GiudGiud** [reply]
hello

mesh refinement could just be added with a `-r n` flag on the command line if you want to try this out

**Reply by SudiptaBiswas** [reply]
This could also depend on what interpolation functions you are using for the diffusivities. Start with a simple function to debug the issue.

**Reply by chongxiyuan** [reply]
Thank you so much! I will start with maybe a linear interpolation of c and etas with finer mesh and keep trying from there.

**Reply by GiudGiud** [reply]
@laimaster21 any luck with this approach?

**Reply by chongxiyuan** [reply]
Hi, sorry for the late reply. I think I found the issue, you were right it is with the interpolation of surface diffusion D_surf. The 2018 paper  https://link.springer.com/article/10.1007/s10853-017-1846-3  equation 32 reports D_surf with c(1-c)D_surf in the C-H mobility. It does not converge very well in my simulations. I used (3*c^2-2*c^3)(1-3*c^2-2*c^3)D_surf in the mobility and it works fine so far. Mobility constants are picked similar scale to the paper.
<img width="721" alt="image" src="https://user-images.githubusercontent.com/35067395/171920007-d566b17e-bb05-48e5-a2d0-77dff204c808.png">
